OpenFeature/src/main/java/dev/openfeature/sdk/async/Extensions.kt

```diff
@@ -0,0 +1,19 @@
+package dev.openfeature.sdk.async
+
+import dev.openfeature.sdk.OpenFeatureClient
+import dev.openfeature.sdk.events.EventHandler
+import dev.openfeature.sdk.events.OpenFeatureEvents
+import dev.openfeature.sdk.events.observe
+import kotlinx.coroutines.flow.onStart
+
+fun OpenFeatureClient.toAsync(): AsyncClient {
+    return AsyncClientImpl(this)
+}
+
+fun observeProviderReady() = EventHandler.eventsObserver()
```

Comment:
Similar comment above: let's extend this to support any ProviderStateEvent type?